packages/protvista-datatable/src/protvista-datatable.ts

```diff
@@ -1,26 +1,79 @@
-import { LitElement, html } from "lit-element";
+import { LitElement, html, TemplateResult, CSSResult } from "lit-element";
 import { v1 } from "uuid";
 import { ScrollFilter } from "protvista-utils";
-/* eslint-disable import/extensions, import/no-extraneous-dependencies */
+
+import { ProtvistaLoadEvent } from "./types/events";
+import { ProtvistaManager } from "./types/manager";
+
 import styles from "./styles";
 
-class ProtvistaDatatable extends LitElement {
+type DataTableDatum = {
+  start?: number;
+  begin?: number;
```

Comment:
I think here you need at least one of these? If so, type-fest has a [require-at-least-one](https://github.com/sindresorhus/type-fest/blob/main/source/require-at-least-one.d.ts) that could enforce that.